tests/python/gstaichi/lang/test_kernel_impl.py

```diff
@@ -45,3 +45,69 @@ def my_cool_test_function(a: ti.types.NDArray[ti.types.vector(2, ti.i32), 2]):
         match=my_cool_test_function.__qualname__,
     ):
         my_cool_test_function(x)
+
+
+@test_utils.test()
+def test_pure_kernel_parameter() -> None:
+    arch = ti.lang.impl.current_cfg().arch
+    ti.init(arch=arch, offline_cache=False, src_ll_cache=True)
+
+    @ti.pure
+    @ti.kernel
+    def k1() -> None: ...
+
+    @ti.kernel(pure=True)
```

Comment:
Do we need tests with a pure kernel and actually do something insdie?

Reply:
Addressed in f7d26946f